src/main/java/org/candlepin/subscriptions/files/PerLineFileSource.java

```diff
@@ -41,24 +42,38 @@
  */
 public class PerLineFileSource implements ResourceLoaderAware {
 
-    private final Cache<List<String>> cache;
+    private final Cache<List<String>> listCache;
+    private final Cache<Set<String>> setCache;
     private ResourceLoader resourceLoader;
     private Resource fileResource;
     private String resourceLocation;
 
     public PerLineFileSource(String resourceLocation, Clock clock, Duration cacheTtl) {
         this.resourceLocation = resourceLocation;
-        this.cache = new Cache<>(clock, cacheTtl);
+        this.listCache = new Cache<>(clock, cacheTtl);
+        this.setCache = new Cache<>(clock, cacheTtl);
     }
 
     public List<String> list() throws IOException {
-        if (cache.isExpired()) {
+        if (listCache.isExpired()) {
             try (InputStream s = fileResource.getInputStream()) {
-                cache.setValue(new BufferedReader(new InputStreamReader(s, Charset.defaultCharset())).lines()
+                listCache.setValue(new BufferedReader(
+                    new InputStreamReader(s, Charset.defaultCharset())).lines()
                     .filter(line -> line != null && !line.isEmpty()).collect(Collectors.toList()));
             }
         }
-        return cache.getValue();
+        return listCache.getValue();
+    }
+
+    public Set<String> set() throws IOException {
+        if (setCache.isExpired()) {
+            try (InputStream s = fileResource.getInputStream()) {
```

Comment:
Could we DRY this a bit like so?
```java
public List<String> list() {
    if (listCache.isExpired() {
        listCache.setValue(coallate());
    } else {
        return listCache;
    }

public Set<String> set() {
    if (setCache.isExpired()) {
        setCache.setValue(new HashSet(coallate()));
    } else {
        return setCache;
    }

protected List<String> coallate() {
    try (InputStream s = fileResource.getInputStream()) {
            return new BufferedReader(
                new InputStreamReader(s, Charset.defaultCharset())).lines()
                    .filter(line -> line != null && !line.isEmpty()).collect(Collectors.toList()));
    } 
...
}
```

Reply:
I did something similar (force pushed to avoid need to rebase):

```diff
diff --git a/src/main/java/org/candlepin/subscriptions/files/PerLineFileSource.java b/src/main/java/org/candlepin/subscriptions/files/PerLineFileSource.java
index 7e62d5b..71f8fd0 100644
--- a/src/main/java/org/candlepin/subscriptions/files/PerLineFileSource.java
+++ b/src/main/java/org/candlepin/subscriptions/files/PerLineFileSource.java
@@ -31,8 +31,10 @@ import java.io.InputStreamReader;
 import java.nio.charset.Charset;
 import java.time.Clock;
 import java.time.Duration;
+import java.util.Collection;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collector;
 import java.util.stream.Collectors;
 
 import javax.annotation.PostConstruct;
@@ -55,25 +57,24 @@ public class PerLineFileSource implements ResourceLoaderAware {
     }
 
     public List<String> list() throws IOException {
-        if (listCache.isExpired()) {
-            try (InputStream s = fileResource.getInputStream()) {
-                listCache.setValue(new BufferedReader(
-                    new InputStreamReader(s, Charset.defaultCharset())).lines()
-                    .filter(line -> line != null && !line.isEmpty()).collect(Collectors.toList()));
-            }
-        }
-        return listCache.getValue();
+        return getCachedValue(listCache, Collectors.toList());
     }
 
     public Set<String> set() throws IOException {
-        if (setCache.isExpired()) {
+        return getCachedValue(setCache, Collectors.toSet());
+    }
+
+    private <T extends Collection<String>> T getCachedValue(Cache<T> cache,
+        Collector<String, ?, T> collector) throws IOException {
+
+        if (cache.isExpired()) {
             try (InputStream s = fileResource.getInputStream()) {
-                setCache.setValue(new BufferedReader(
+                cache.setValue(new BufferedReader(
                     new InputStreamReader(s, Charset.defaultCharset())).lines()
-                    .filter(line -> line != null && !line.isEmpty()).collect(Collectors.toSet()));
+                    .filter(line -> line != null && !line.isEmpty()).collect(collector));
             }
         }
-        return setCache.getValue();
+        return cache.getValue();
     }
 
     @Override
```

Reply:
Love it.